lang/py/src/avro/schema.py

```diff
@@ -405,6 +420,106 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale):
+    LogicalSchema.__init__(self, decimal_type, 'decimal')
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+
+  def validate(self, validation_props=None):
```

Comment:
I have revamped the LogicalSchema design. The previous one had some issues with it.